lib/active_record/connection_adapters/oracle_enhanced_adapter.rb

```diff
@@ -763,6 +763,23 @@ def bind_string(name, value)
           ActiveRecord::Relation::QueryAttribute.new(name, value, Type::OracleEnhanced::String.new)
         end
 
+        # call select_values using binds even if surrounding SQL preparation/execution is done
+        # with conn.unprepared_statement (like AR.to_sql)
+        def select_values_forcing_binds(arel, name, binds)
+          # remove possible force of unprepared SQL during dictionary access
+          unprepared_statement_forced = prepared_statements_disabled_cache.include?(object_id)
```

Comment:
I did not know `prepared_statements_disabled_cache` method implemented via  https://github.com/rails/rails/commit/206345b9890